packages/graph/hash_graph/lib/graph/src/api/rest/data_type.rs

```diff
@@ -118,6 +120,24 @@ async fn create_data_type<P: StorePool + Send>(
         .map(Json)
 }
 
+#[utoipa::path(
+    post,
+    path = "/data-types/query",
+    request_body = Expression,
+    tag = "DataType",
+    responses(
+        (status = 200, content_type = "application/json", description = "List of all data types matching the provided query", body = [PersistedDataType]),
+
+        (status = 500, description = "Store error occurred"),
+    )
+)]
+async fn get_data_types_by_query<P: StorePool + Send>(
```

Comment:
I don't think it's really important, and we shouldn't block the PR on it (in my opinion), but:

Should we think about renaming some of these to `read_*` or `retrieve_*` or something? It's a little strange to call it a `get` if it's a `post`

Reply:
Applied in affe0f026cd75cc7464705857f2902d0a3b8d632

Reply:
Hm well now it's inconsistent with some of the other ones 🙃 

I think we should either revert the commit and discuss as a team if we want to do this, or we should update all of the other `get_` methods to be called something else. (I'm not sure I'm a huge fan of 'query')

Reply:
I reverted the commit so we can take care of this later to unblock this PR.